Installation issue: hdf-eos2 failing to build with ^hdf+szip

When trying to build hdf-eos2 using an hdf built with +szip, I get an error during configure
about lack of an --with-szlib flag.

It also looks like the checks for jpeg and zlib libraries will fail if hdf was not built with them
(e.g. using ``` if self.spec['jpeg']:``` instead of ``if 'jpeg' in self.spec:```)

### Steps to reproduce the issue

```console
$ spack install hdf-eos2 ^hdf+szip
...
configure: error: HDF4 was linked with SZIP, but --with-szlib was not given
...
```

### Information on your system

```console
$ spack debug report
* **Spack:** 0.16.1-2431-da8039bf29
* **Python:** 3.6.8
* **Platform:** linux-rhel8-x86_64
* **Concretizer:** original
```
### Additional information

* [spack-build-out.txt](https://github.com/spack/spack/files/6400592/spack-build-out.txt)
* [spack-build-env.txt](https://github.com/spack/spack/files/6400593/spack-build-env.txt)

No maintainers for hdf-eos2, but IIRC I (@payerle) am the one who uploaded the original version.

### General information

- [x ] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x ] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x ] I have uploaded the build log and environment files
- [x ] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hdf-eos2/package.py
+++ b/var/spack/repos/builtin/packages/hdf-eos2/package.py
@@ -86,10 +86,13 @@ class HdfEos2(AutotoolsPackage):
 
         # Provide config args for dependencies
         extra_args.append('--with-hdf4={0}'.format(self.spec['hdf'].prefix))
-        if self.spec['jpeg']:
+        if 'jpeg' in self.spec:
             extra_args.append('--with-jpeg={0}'.format(
                 self.spec['jpeg'].prefix))
-        if self.spec['zlib']:
+        if 'libszip' in self.spec:
+            extra_args.append('--with-szlib={0}'.format(
+                self.spec['libszip'].prefix))
+        if 'zlib' in self.spec:
             extra_args.append('--with-zlib={0}'.format(
                 self.spec['zlib'].prefix))
 

